src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -2088,7 +2088,7 @@ public static void uploadWithoutConnection(SnowflakeFileTransferConfig config) t
           break;
       }
     } catch (Exception ex) {
-      logger.error("Exception encountered during file upload: ", ex.getMessage());
+      logger.error("Exception encountered during file upload.", ex);
```

Comment:
We don't need to expose whole exception as we need to make sure it doesn't have any credential information. Do you have test which actually don't have error message? Then we can fix the cause of it and add error message where ever needed.

Reply:
Yes, it's common in our logs to get the following:

```
Exception encountered during file upload: 
```

I can't tell you where it come from because there is no message or logging information :) If I had to guess there is an IOException or similar being thrown without a message, but its impossible to know because the exception could be from anywhere in the callstack. I could enable debug logs to maybe get more detail, but I shouldn't have to.

> as we need to make sure it doesn't have any credential information.

This seems overly defensive, wouldn't the message be most likely to contain the credential information? Wouldn't you explicitly not want to log the thing that is user controlled (from the codes perspective). I don't see how a stack trace would have credential information, but I'm probably wrong. Interestingly, why do you log the full exception on this line then?

https://github.com/snowflakedb/snowflake-jdbc/blob/ab313969251c215208f30175a84051aba5fe5baa/src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java#L608C11-L608C72

It would seem like user error and also incredibly silly to put credential information in an exception

Reply:
ok. fair enough. Can you create an issue and link this PR to it? Normally PR always associate with an issue.

Reply:
Done!